pydantic/type_adapter.py

```diff
@@ -203,36 +240,105 @@ def __init__(
                 code='type-adapter-config-unused',
             )
 
-        config_wrapper = _config.ConfigWrapper(config)
-
-        core_schema: CoreSchema
+        self._type = type
+        self._config = config
+        self._parent_depth = _parent_depth
+        if module is None:
+            f = sys._getframe(1)
+            self._module_name = cast(str, f.f_globals.get('__name__', ''))
+        else:
+            self._module_name = module
+
+        self._core_schema: CoreSchema | None = None
+        self._validator: SchemaValidator | None = None
+        self._serializer: SchemaSerializer | None = None
+
+        if not self._defer_build():
+            # Immediately initialize the core schema, validator and serializer
```

Comment:
This is a bit fuzzy to me still. Core schema and friends are **not** initialized immediately when `_type` is subclass of `BaseModel` even though `_defer_build_mode` does not include `type_adapter`. This is how it also behaved previously and we probably can not really change it.

But the case it breaks (or is inconsistent) is/was when `BaseModel` is inside `Annotated[BaseModel, Field]`. I feel like this is a bit strange behavior even in current state of PR. So the Annotated wrapping case suddenly requires including the `type_adapter` to `_defer_build_mode` even though it includes `model`. The behavior is quite a best to document properly. The new `type_adapter` is quite fitting for rest of TypeAdapter types but the Annotated is strange that mode is needed for normal BaseModels.

I would say most clear way would be to just allow deferred building for the Annotated case when `_defer_build_mode` includes just `model`. But then again there were some conserns with this...